src/definition_provider/modelDefinitionProvider.ts

```diff
@@ -12,16 +12,18 @@ import {
   Disposable,
 } from "vscode";
 import { NodeMetaMap } from "../domain";
-import { dbtProjectContainer } from "../manifest/dbtProjectContainer";
+import { DbtProjectContainer } from "../manifest/dbtProjectContainer";
 import { ManifestCacheChangedEvent } from "../manifest/event/manifestCacheChangedEvent";
+import { provideSingleton } from "../utils";
 
+@provideSingleton(ModelDefinitionProvider)

```

Comment:
The question I ask myself is do we need to set it as provideSingleton if we only use it in one place?